src/V3Param.cpp

```diff
@@ -131,6 +153,13 @@ class ParameterizedHierBlocks final {
                     AstConst* constp = VN_CAST(pinp->exprp(), Const);
                     UASSERT_OBJ(constp, pinp,
                                 "parameter for a hierarchical block must have been constified");
+                    AstConst* defValue = VN_CAST(
+                        findWithCheck(defaultValues, modvarp->name(), modvarp)->valuep(), Const);
+                    if (defValue && areSame(constp, defValue)) {
+                        UINFO(5, "Setting default value of " << constp << " to " << modvarp
+                                                             << std::endl);
+                        continue;  // Skip this parameter because setting the same value
+                    }
```

Comment:
Wonder if we should instead make an earlier pass to just remove any parameters that set to defaults?  Ok either way.

Reply:
Yes. That is also one option I considered too.
I chose here so as to collect hier_block specific code in this function.